icaruscode/TPC/Tracking/cluster3D/SnippetHit3DBuilderICARUS_tool.cc

```diff
@@ -1646,12 +1644,9 @@ geo::WireID SnippetHit3DBuilderICARUS::NearestWireID(const Eigen::Vector3f& posi
     try
     {
         // Switch from NearestWireID to this method to avoid the roundoff error issues...
-        //double distanceToWire = m_geometry->Plane(wireIDIn).WireCoordinate(geo::vect::toPoint(position.data()));
-
-        //wireID.Wire = int(distanceToWire);
+        double distanceToWire = m_wireReadoutAlg->Plane(wireIDIn).WireCoordinate(geo::vect::toPoint(position.data()));
 
-        // Not sure the thinking above but is wrong... switch back to NearestWireID...
-        wireID = m_geometry->NearestWireID(geo::vect::toPoint(position.data()),wireIDIn);
+        wireID.Wire = int(distanceToWire);
```

Comment:
My reading of the code is that the former code was using rounding to the closest integer (inside `NearestWireID()`), while this is explicitly using truncation.
In addition, the original code can throw a (annoying) exception (`geo::InvalidWireError`, as described in the [documentation](https://code-doc.larsoft.org/docs/latest/html/classgeo_1_1PlaneGeo.html#a0f2906b4332d3eedc05c2191879a1ae4)), while here this will not happen.
That documentation also suggests a possible solution. The author of this code (@SFBayLaser?) should weigh in.

Reply:
Yes, the TODO in  the documentation would be an ideal solution!

Reply:
Ok, I  went back to the original file to remind myself... there is a nearly decade long history here (from the early days of cluster3d) where there were issues, obviously, using the WireCoordinate method which were not understood (hence it being commented out). I am pretty sure this last configuration has been in place for around 5-6 years and I'd bet that now it could be done better... 
I'd accept the change for now, then we should compare this file against the one in sbncode, reconcile differences and then remove this from icaruscode. (and, really, we should update everything in larreco so it can be used for DUNE too since it seems someone has been picking it up).